lib/sneakers/worker.rb

```diff
@@ -92,6 +92,16 @@ def do_work(delivery_info, metadata, msg, handler)
 
         metrics.increment("work.#{self.class.name}.ended")
       end #process
+    rescue => ex
+      # If the thread pool is shutting down, it will throw a RuntimeError
+      # So, requeue the message as it is not processed.
+      handler.reject(delivery_info, metadata, msg, true)
+
+      # If is was not the thread pool shutting down, let's know about the error:
+      unless ( ex.to_s =~ /unable to add work while shutting down/i )
```

Comment:
First, it's not Bunny, but Thread, throwing the error.

Second, I don't think this is a dependency. It's more of a service.
We expect Thread to throw an error, let's not make noise if it's the error we expect.

I do not see a big difference in matching on a class or a string... Both are quite fragile and depending on the underlying gem to actually throw exactly that.

Anyhow, I will investigate if messages are dropped, if passed to Sneakers by Bunny while shutting down. I have a suspicion that they actually are. That would be, sorry that I'm repeating myself, critical.